### ISSUE TITLE
Service injected methods are convenient

### ISSUE THREAD

**Issue by eduardoarandah** [CrestApps/laravel-code-generator#81]
Hey! 
Just a little feedback:

1) Methods are better when initialized automatically with model, 

instead of 

```

public function edit($id){
$company = Company::findOrFail($id);


```

better:

````

public function edit(Company $company){
$company;

````

Why?

Autorization via policies rely on service injected controllers: 
https://laravel.com/docs/5.6/authorization#creating-policies

This is an example of policy-authorizing a route:

```

Route::get('/{company}', 'CompanyController@show')
            ->name('show')
            ->where('id', '[0-9]+')
            ->middleware('can:show,company');
```

2) You have try-catch methods that don't log anything, making it hard to detect errors

Here in line 110 I recommend "report" method
https://github.com/CrestApps/laravel-code-generator/blob/master/templates/default/controller.stub

like this:
```

try {
        // Validate the value...
    } catch (Exception $e) {
        report($e);

....
    }


```

**Comment by CrestApps** [comment]
Agreed it is nicer. I think this is something Laravel 5.3+ supports. I'll put it on the todo list.
